File: src-tauri/src/commands/keymap.rs
use crate::services;
use serde::Deserialize;

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WriteKeymapParams {
    pub contents: String,
}

#[tauri::command]
pub fn read_keymap(app: tauri::AppHandle) -> Result<String, String> {
    services::keymap::read(&app)
}

#[tauri::command]
pub fn write_keymap(app: tauri::AppHandle, params: WriteKeymapParams) -> Result<(), String> {
    services::keymap::write(&app, params.contents)
}

File: src/lib/api/keymap.ts
import { invokeCommand, type InvokeOptions } from "./utils.ts";

export function readKeymap(options?: InvokeOptions): Promise<string> {
  return invokeCommand<string>("read_keymap", undefined, options);
}

export type WriteKeymapParams = { contents: string };

export async function writeKeymap(
  params: WriteKeymapParams,
  options?: InvokeOptions,
): Promise<void> {
  await invokeCommand<void>("write_keymap", params, options);
}
